src/scripts/main.js

```diff
@@ -1,3 +1,38 @@
 'use strict';
 
-// write your code here
+const logo = document.querySelector('.logo');
+
+function createPromise() {
+  const resolver = (resolve, reject) => {
+    logo.addEventListener('click', () => {
+      resolve();
+    });
+  };
+
+  return new Promise(resolver);
+}
+
+const promise1 = createPromise();
+
+promise1.then(() => {
+  success(success);
+
+  const promise2 = new Promise((resolve, reject) => {
+    setTimeout(() => reject(error), 3000);
+  });
+
+  return promise2;
+}, error)
+  .then(success, error);
```

Comment:
It's better to handle onError in catch block because you may have any unexpected errors not only rejected promises.